sdks/java/io/elasticsearch/src/main/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIO.java

```diff
@@ -2838,9 +2850,14 @@ private List<Document> flushBatch() throws IOException, InterruptedException {
           responseEntity = new BufferedHttpEntity(response.getEntity());
         } catch (java.io.IOException ex) {
           if (spec.getRetryConfiguration() == null || !isRetryableClientException(ex)) {
-            throw ex;
+            if (spec.getThrowWriteErrors()) {
```

Comment:
@andyzhangdialpad I'm considering whether it makes sense to have yet another method along the lines of `isPermanentlyRetryable` that could handle cases like HTTP 503, and using it like:

```java
if (spec.getThrowWriteErrors() || isPermanentlyRetryable(ex)) {
  throw ex;
}
```

This would allow for recovering an ES cluster from a bad state and letting ingestion resume without the need for additional processing of `FAILED_WRITES` 🤔

Reply:
Having fewer "special cases" of error handling is likely a good thing, I agree.  With the proposed changes in the PR as of now, a 503 would be retried according to RetryConfiguration, which feels right/predictable.  

IMO this can be merged once tests pass.